content/post/language_migrations.md

```diff
@@ -0,0 +1,512 @@
+</div>
+</div>
+
+* Developers coding in one of the 5 most popular languages (**Java**, **C**, **C++**, **PHP**, **Ruby**) are more likely to switch to **Python** with around that 22 % of chance in average.
+
+* Also according to Erik's matrix, in case we use **Ojective-C**, we would switch to **Swift** and reversely with the greatest probability (24 and 19 %).
+
+* Similarly, a **Visual Basic** developer has more chance (24 %) to move to **C#**, when Erik's reaches the nearly certitude with 92 %.
+
+* GitHub users of **Clojure**, **C#** and above all **Scala** would rather switch to **Java** with respectively 22, 29 and 40 % of chance which is the highest probability.
+
+* Most of people using numerical and statistical environments such as **Fortran** (36 %), **Matlab** (33 %) or **R** (40 %), move to **Python** unlike Erik's matrix which predicts **C** as their future language.
+
+* One common point I found with Erik's results about **golang** is that relatively to the previous language, the highest proportion of people moving to **Go**, belongs to people giving up with **Rust** (9 % for our matrix and 47 % for Erik's).
+
+## Over the last 16 years
+
+Now, focusing on a specific year, what do the yearly transition matrices look like ? Do they express the same find of language profile ? How has it been evolving since the early 20's ? Here is a sample of 4 matrices to feel the time-line.
+
+<div class="grid2x">
+<div>
+<div>
+<img src="/post/language_migrations/P_2005.svg">
+<p align="center" class="dt"><pre><code class="hljs python">2005 - 2006</code></pre></p>
+</div>
+<div>
+<img src="/post/language_migrations/P_2011.svg">
+<p align="center" class="dt"><pre><code class="hljs python">2011 - 2012</code></pre></p>
+</div>
+</div>
+<div>
+<div>
+<img src="/post/language_migrations/P_2008.svg">
+<p align="center" class="dt"><pre><code class="hljs python">2008 - 2009</code></pre></p>
+</div>
+<div>
+<img src="/post/language_migrations/P_2015.svg">
+<p align="center" class="dt"><pre><code class="hljs python">2015 - 2016</code></pre></p>
+</div>
+</div>
+</div>
+
+<p align="center" class="caption">Temporal evolution of transition matrices</p>
+
+In the end, the evolution of those matrices over time seems to be affine, and we detect the same language profile every year. Thus, to highlight the time-line of this language profile, we apply the power iteration to each matrix. Earlier, we averaged the stationary distribution over the last 16 years but now we look at its chronological sequence. The resulting series of dominant eigenvectors is presented below in a **stack area plot**.
+
+<img src="/post/language_migrations/eigenvect_stack_22lang.png" class="fig">
+<p align="center" class="caption">Stationary distribution of languages over the last 16 years</p>
+
+Each language is represented by a layer whose thickness is equal to its corresponding value in the dominant eigenvector for a given year. Besides, it is sorted by the popularity we studied earlier.
```

Comment:
The thickness of each band corresponds to the value in the dominant eigenvector. The bands are sorted by the averaged popularity which we calculated before.